coverage.py

```diff
@@ -89,7 +92,8 @@ def __init__(self, collection, uploader=None,
         # Since we are the metadata wrangler, any resources we find,
         # we mirror to S3.
         if not uploader:
-            uploader = S3Uploader()
+            uploader = S3Uploader.from_config(self._db)
```

Comment:
I have a branch that changes `S3Uploader.from_config()` to return None instead of raising an exception if no S3 integration is defined. In the circ manager this is the right thing to do, but in metadata and content you need to check that an S3Uploader is actually returned, and raise `CannotLoadConfiguration` if not.